### DISCUSSION TITLE
Changing preferred sources

### DISCUSSION THREAD

**Discussion by plyoung** [ccatterina/plasmusic-toolbar#101] [General]
Is there a way to add my own preferred sources ? I only see any, spotify, and vlc.
It works with Audacious in "any" source but then it also picks up my browser playing some youtube, which I don't want.

**Comment by ccatterina** [comment]
Hi!
Yes, you need to type the application name in the preferred source dropdown, press enter, and then click on apply botton.
To find the application name you can look in `/usr/share/applications/`, the name you need to type in the dropdown is the same as the `.desktop` file (without the .desktop extension).

You can find a gif here: https://github.com/ccatterina/plasmusic-toolbar/issues/91#issuecomment-2050037738

I have to change this mecanism to make it more user friendly or at least explain how to use it in the readme!

**Reply by plyoung** [reply]
Ooh, did not realise I could type in there. Thanks :D

**Comment by pop9459** [comment]
Hi, I'm having a similar issue. I'm using Tidal as a kde webapp but the problem is the .desktop entry is not in `/usr/share/applications/`. I found it in `/home/username/.local/share/applications/webapp-Tidal2200.desktop` but setting `webapp-Tidal2200` as the preferred source doesn't work.
I also tried moving the .desktop file to `/usr/share/applications/` or setting `chromium-browser` as the source but without any luck.
If I use "any" it works so I'm sure there is a way. Does anybody know about a solution to this?

**Reply by ccatterina** [reply]
Hi!
what do you mean with "kde webapp"? Do you open the tidal app in the browser (https://listen.tidal.com/)?
If yes, you should probably look for the browser `.desktop` file name. Otherwise can you share a link of the app you are using? I can try to install it and make some tests. :slightly_smiling_face:

**Reply by pop9459** [reply]
Hey, thanks for the reply.
I'm actually not sure if this is a kde specific thing but my nobara 39 install comes with a utility called webApps. It basically allows you to "Run websites as if they were apps". You create an entry, lets say for Tidal and set the address to https://listen.tidal.com/. Then, when want to start it, it will open an isolated browser window for that "app".

![image](https://github.com/ccatterina/plasmusic-toolbar/assets/71280916/387eec69-d88a-4261-b0ca-9f8a9a0abcb8)
![image](https://github.com/ccatterina/plasmusic-toolbar/assets/71280916/642a5d1e-9ddb-4938-a487-84d150a0192b)

I can't find a repo or a wiki for it anywhere but here are the packages from my system that match the name. 

webapp-manager.noarch                                                                    master.mint21-1.fc39                                             @nobara-baseos                                 
jetty-webapp.noarch                                                                      9.4.40-8.fc39                                                    fedora                                         
tomcat-admin-webapps.noarch                                                              1:9.0.83-1.fc39                                                  updates                                        
tomcat-docs-webapp.noarch                                                                1:9.0.83-1.fc39                                                  updates                                        
tomcat-webapps.noarch                                                                    1:9.0.83-1.fc39                                                  updates                                        
webapp-manager.src                                                                       master.mint21-1.fc39                                             nobara-baseos-39

**Reply by ccatterina** [reply]
Hi! sorry for the late reply.
I found  the app you are using to create web apps, it is https://github.com/linuxmint/webapp-manager.
Based on how it works, the source to use should be the browser name, `chromium` in your case. Unfortunately I made some test on my pc using chromium and setting `chromium` as source but it doesn't work. It seems that the `desktopEntry` mpris parameter that I use as a discriminant to switch between players is not populated for chromium and chromium-based browsers.
I will investigate a bit more in the next few days and I let you know.

**Reply by pop9459** [reply]
Thx for investigating

**Reply by ccatterina** [reply]
Hi! I confirm that in chromium `desktopEntry` MPRIS property is not implemented (https://github.com/chromium/chromium/blob/87743c655c2127d9a73b84620b76e1b9dfcf455c/components/system_media_controls/linux/system_media_controls_linux.cc#L261-L291).

Maybe I could add an other optional configuration that allow to choose preferred source using MPRIS `Identity` peoperty. I can't think of any other way.

**Reply by ccatterina** [reply]
https://github.com/ccatterina/plasmusic-toolbar/issues/107

**Reply by ccatterina** [reply]
@pop9459 
With the new preferred source configuration released with v2.0.0, you should be able to select Chromium as the preferred source.
